hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -173,6 +189,49 @@ private boolean deleteArchivedInstants(List<HoodieInstant> archivedInstants) {
         throw new HoodieIOException("Failed to delete archived instant " + archivedInstant, e);
       }
     }
+
+    // Remove older meta-data from auxiliary path too
+    Optional<HoodieInstant> youngestCommitted =
+        archivedInstants.stream()
+            .filter(i -> {
+              return i.isCompleted()
+                  && (i.getAction().equals(COMMIT_ACTION) || (i.getAction().equals(DELTA_COMMIT_ACTION)));
+            })
+            .sorted(Comparator.comparing(HoodieInstant::getTimestamp).reversed()).findFirst();
+    if (youngestCommitted.isPresent()) {
+      success &= deleteAllInstantsOlderorEqualsInAuxMetaFolder(youngestCommitted.get());
+    }
+    return success;
+  }
+
+  /**
+   * Remove older instants from auxiliary meta folder
+   *
+   * @param instant Hoodie Instant
+   * @return success if all eligible file deleted successfully
+   * @throws IOException in case of error
+   */
+  private boolean deleteAllInstantsOlderorEqualsInAuxMetaFolder(HoodieInstant instant)
```

Comment:
Have we added tests for this ?

Reply:
+1

Reply:
New Test cases and validations are added in TestHoodieCommitArchiveLog